chartfx-dataset/src/main/java/de/gsi/dataset/spi/CircularDoubleErrorDataSet.java

```diff
@@ -206,6 +229,23 @@ public CircularDoubleErrorDataSet reset() {
 
     @Override
     public DataSet set(final DataSet other, final boolean copy) {
-        throw new UnsupportedOperationException("copy setting transposed data set is not implemented");
+        lock().writeLockGuard(() -> other.lock().writeLockGuard(() -> {
```

Comment:
Why is the write lock (as opposed to a read lock) on `other` necessary? From logic I would say that other is not changed, so other threads should be able to read simultaneously? The tests also pass if I change it to `other.lock().readLockGuard(...)` but I think it might be necessary to prevent deadlocking when the code inside the guard needs a write lock.  If it is necessary, maybe add a small comment with a short explanation, so it will not be carelessly removed in the future?
As this is consistent through the whole codebase, I think we can also take a look at this separately and leave this PR for the fix of the reported issue.

Reply:
Otherwise `dataset.set(dataset)` will deadlock because the read  lock cannot be upgraded to a write lock. This can also happen indirectly e.g. when using dataset wrappers.